netsync/manager.go

```diff
@@ -393,6 +393,18 @@ func (sm *SyncManager) handleDonePeerMsg(peer *peerpkg.Peer) {
 
 	log.Infof("Lost peer %s", peer)
 
+	sm.clearRequestedState(state)
+
+	if peer == sm.syncPeer {
+		// Update the sync peer and disconnect our current sync peer.
+		sm.updateSyncPeer(true)
```

Comment:
I think this can just be `false` since the peer has already been disconnected at this point.